test/unit/org/apache/cassandra/cql3/validation/operations/InsertTest.java

```diff
@@ -247,4 +252,42 @@ public void testCKInsertWithValueOver64K() throws Throwable
         assertInvalidThrow(InvalidRequestException.class,
                            "INSERT INTO %s (a, b) VALUES ('foo', ?)", new String(TOO_BIG.array()));
     }
+
+    @Test
+    public void testInsertingMapDataWithParameterizedQueriesIsKeyOrderIndependent() throws Throwable
+    {
+        UUID uuid1 = UUIDs.timeBased();
+        UUID uuid2 = UUIDs.timeBased();
+        createTable("CREATE TABLE %s (k text, c frozen<map<timeuuid, text>>, PRIMARY KEY (k, c));");
```

Comment:
The fix is for both maps and sets but the tests only exercise maps. I think we should add equivalent tests for sets. for example:
```java
@Test
public void testInsertingSetDataWithParameterizedQueriesIsKeyOrderIndependent() throws Throwable
{
    UUID uuid1 = UUIDs.timeBased();
    UUID uuid2 = UUIDs.timeBased();
    createTable("CREATE TABLE %s (k text, c frozen<set<timeuuid>>, PRIMARY KEY (k, c));");
    // We use `executeNet` in this test because `execute` passes parameters through CqlTester#transformValues(), which calls
    // AbstractType#decompose() on the value, which "fixes" the map order, but wouldn't happen normally.
    executeNet("INSERT INTO %s (k, c) VALUES ('0', ?)", set(uuid1, uuid2));
    executeNet("INSERT INTO %s (k, c) VALUES ('0', ?)", set(uuid2, uuid1));
    beforeAndAfterFlush(() -> {
        assertRowCountNet(executeNet("SELECT * FROM %s WHERE k='0' AND c={" + uuid1 + ", " + uuid2 + '}'), 1);
        assertRowCountNet(executeNet("SELECT * FROM %s WHERE k='0' AND c={" + uuid2 + ", " + uuid1 + '}'), 1);
        assertRowCountNet(executeNet("SELECT * FROM %s WHERE k='0' AND c={" + uuid1 + ", " + uuid2 + '}'), 1);
        assertRowCountNet(executeNet("SELECT * FROM %s WHERE k='0' AND c={" + uuid2 + ", " + uuid1 + '}'), 1);
    });
}

@Test
public void testInsertingSetDataWithParameterizedQueriesIsKeyOrderIndependentWithSelectSlice() throws Throwable
{
    UUID uuid1 = UUIDs.timeBased();
    UUID uuid2 = UUIDs.timeBased();
    createTable("CREATE TABLE %s (k text, c frozen<set<timeuuid>>, PRIMARY KEY (k, c));");
    // We use `executeNet` in this test because `execute` passes parameters through CqlTester#transformValues(), which calls
    // AbstractType#decompose() on the value, which "fixes" the map order, but wouldn't happen normally.
    beforeAndAfterFlush(() -> {
        executeNet("INSERT INTO %s (k, c) VALUES ('0', ?)", set(uuid2, uuid1));
        assertRowsNet(executeNet("SELECT k, c[" + uuid2 + "] from %s"), row("0", uuid2));
        assertRowsNet(executeNet("SELECT k, c[" + uuid1 + "] from %s"), row("0", uuid1));
    });
}
```

Reply:
Interestingly, sets already worked because they already used `SortedSet` everywhere... that said, happy to add some tests if we want to make sure we don't later _break_ sets somehow... will wait on the quesiton on where to put them (here or Collections.java) before implementing though.